airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java

```diff
@@ -164,44 +175,27 @@ protected BigQuery getBigQuery(final JsonNode config) {
     }
   }
 
-  private ServiceAccountCredentials getServiceAccountCredentials(JsonNode config) throws IOException {
-    ServiceAccountCredentials credentials;
-    final String credentialsString =
-        config.get(BigQueryConsts.CONFIG_CREDS).isObject() ? Jsons.serialize(config.get(BigQueryConsts.CONFIG_CREDS))
-            : config.get(BigQueryConsts.CONFIG_CREDS).asText();
+  private ServiceAccountCredentials getServiceAccountCredentials(final JsonNode config) throws IOException {
+    final ServiceAccountCredentials credentials;
+    final String credentialsString = config.get(BigQueryConsts.CONFIG_CREDS).isObject()
+        ? Jsons.serialize(config.get(BigQueryConsts.CONFIG_CREDS))
+        : config.get(BigQueryConsts.CONFIG_CREDS).asText();
     credentials = ServiceAccountCredentials
         .fromStream(new ByteArrayInputStream(credentialsString.getBytes(Charsets.UTF_8)));
     return credentials;
   }
 
-  /**
-   * Strategy:
-   * <p>
-   * 1. Create a temporary table for each stream
-   * </p>
-   * <p>
-   * 2. Write records to each stream directly (the bigquery client handles managing when to push the
-   * records over the network)
-   * </p>
-   * <p>
-   * 4. Once all records have been written close the writers, so that any remaining records are
-   * flushed.
-   * </p>
-   * <p>
-   * 5. Copy the temp tables to the final table name (overwriting if necessary).
-   * </p>
-   *
-   * @param config - integration-specific configuration object as json. e.g. { "username": "airbyte",
-   *        "password": "super secure" }
-   * @param catalog - schema of the incoming messages.
-   * @return consumer that writes singer messages to the database.
-   */
   @Override
   public AirbyteMessageConsumer getConsumer(final JsonNode config,
                                             final ConfiguredAirbyteCatalog catalog,
                                             final Consumer<AirbyteMessage> outputRecordCollector)
       throws IOException {
-    return getRecordConsumer(getUploaderMap(config, catalog), outputRecordCollector);
+    final UploadingMethod uploadingMethod = BigQueryUtils.getLoadingMethod(config);
+    if (uploadingMethod == UploadingMethod.STANDARD) {
+      return getStandardRecordConsumer(config, catalog, outputRecordCollector);
```

Comment:
BTW, should there be some warning printed in the logs to warn users and recommend using staging options rather than direct upload methods though? (especially when dealing with larger streams)

Reply:
I guess it's always staging in avro...

Reply:
Will do. I will also update the document to mention that.